### DISCUSSION TITLE
[Turborepo] Cache individual pages' compilation in `next dev`

### DISCUSSION THREAD

**Discussion by oliverjohns** [vercel/turborepo#3156] [Ideas]
I think this would be great. Ideally when running `next dev` and you route to a page and it is compiled, turborepo can store a hash for this specific page and only invalidate cache if that page or its dependencies change.

I am not sure how this would be implemented in practice, but I think its worth considering since it would speed up development. For example, in our next app each page takes roughly 20 seconds to compile, and caching those compilations would be of tremendous help.

Since caching is done "per task", maybe each page compilation could be considered its own task in order to adhere to the current setup for turborepo?

**Comment by anthonyshew** [comment]
What you're describing is the goal of Turbopack!
